### PR TITLE
feat: add service list command

### PR DESCRIPTION
**Pull request by zccabb**
+ fix Error after executing make codegen in resource/list_test
+ feat:add service list command

### CODE REVIEW

File: internal/options/types.go [hunk 1]
```diff
@@ -193,6 +195,18 @@ type ResourceOptions struct {
 	// Get specifies resource.
 	Get GetOption
 }
+type ServiceOptions struct {
```
Comment (tokers) [inline internal/options/types.go:198]: Ditto, all of the structures are not necessary.

File: cmd/service/list.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+					output.Errorf("Failed to list services: %s", err.Error())
+				}
+				json, _ := json.MarshalIndent(servicesList, "", "\t")
+				fmt.Println(string(json))
+				return
+			}
+			output.Errorf("This kind of resource is not supported")
+		},
+	}
+	cmd.PersistentFlags().StringVar(&options.Global.Service.List.Kind, "kind", "service", "Specify the resource kind")
```
Comment (mscb402) [inline cmd/service/list.go:60]: I think `service` command is only used to list services. What `kind` used for?

File: internal/cloud/api.go [hunk 3]
```diff
@@ -160,6 +160,42 @@ func (a *api) GetClusterDetail(clusterID cloud.ID) (*cloud.Cluster, error) {
 	return cluster, nil
 }
 
+func (a *api) ListServices(clusterID cloud.ID, limit int, skip int) ([]*cloud.Application, error) {
+	var services []*cloud.Application
+	pageSize := limit
+	page := int(math.Floor(float64(skip)/float64(limit))) + 1
+	start := skip % limit
+	end := skip%limit + limit
```
Comment (tokers) [inline internal/cloud/api.go:168]: It looks complicated.

Why not use a fixed page size, e.g., 25

Then:

```
startPage := skip / pageSize
startPageSkip = pageSize - skip % pageSize
```

File: cmd/service/list.go (hunk 2 above)
Comment (zccabb) [reply cmd/service/list.go]: fixed

File: cmd/service/cmd.go
[the remark this answers is not in the document]
Comment (zccabb) [reply cmd/service/cmd.go:1]: fixed

File: cmd/service/list.go
[the remark this answers is not in the document]
Comment (zccabb) [reply cmd/service/list.go:1]: fixed

File: internal/options/types.go (hunk 1 above)
Comment (zccabb) [reply internal/options/types.go]: fixed

File: internal/cloud/api.go (hunk 3 above)
Comment (zccabb) [reply internal/cloud/api.go]: fixed

File: cmd/resource/list.go [hunk 4]
```diff
@@ -47,18 +47,32 @@ func newListCommand() *cobra.Command {
...
-			user, err := cloud.Client().Me()
-			if err != nil {
-				output.Errorf(err.Error())
-			}
 			if kind == "cluster" {
+				user, err := cloud.Client().Me()
+				if err != nil {
+					output.Errorf(err.Error())
+				}
+
```
Comment (mscb402) [inline cmd/resource/list.go:55]: This `cloud.Client().Me()` should not move to inside.

File: cmd/resource/list.go [hunk 5]
```diff
@@ -47,18 +47,32 @@ func newListCommand() *cobra.Command {
...
 				if err != nil {
 					output.Errorf("Failed to list clusters: %s", err.Error())
 				}
 				json, _ := json.MarshalIndent(clustersList, "", "\t")
 				fmt.Println(string(json))
 				return
+			} else if kind == "service" {
+				cluster, err := cloud.Client().GetDefaultCluster()
+				if err != nil {
+					output.Errorf(err.Error())
```
Comment (mscb402) [inline cmd/resource/list.go:66]: add some error message

File: internal/cloud/api.go [hunk 6]
```diff
@@ -160,6 +160,42 @@ func (a *api) GetClusterDetail(clusterID cloud.ID) (*cloud.Cluster, error) {
...
+		if err != nil {
+			return nil, err
+		}
+		if service == nil || len(services) == limit {
+			if len(services) > startPageSkip {
+				return services[startPageSkip:limit], nil
+			}
+			return services[:limit], nil
+		}
+		services = append(services, service)
```
[the remark this answers is not in the document]
Comment (zccabb) [reply internal/cloud/api.go:194]: It is suggested that I modify it this way.
 https://github.com/api7/cloud-cli/pull/105#discussion_r1176154910

File: cmd/resource/list.go (hunk 5 above)
Comment (zccabb) [reply cmd/resource/list.go]: Done

File: cmd/resource/list.go (hunk 4 above)
Comment (zccabb) [reply cmd/resource/list.go]: Done

File: internal/cloud/api.go (hunk 3 above)
Comment (zccabb) [reply internal/cloud/api.go:168]: fixed

**Review by mscb402 — APPROVED** [review]

File: internal/options/types.go [hunk 7]
```diff
@@ -192,6 +192,12 @@ type ResourceOptions struct {
 	List ListOption
 	// Get specifies resource.
 	Get GetOption
+
+	Delete DeleteOption
```
Comment (tokers) [inline internal/options/types.go:196]: Remove these useless fields.

Comment (zccabb) [reply internal/options/types.go]: Done

**Review by tokers — APPROVED** [review]

**Pull request merged**